### DISCUSSION TITLE
Edge侧边栏老是切断对话，忍不了了！准备动手！

### DISCUSSION THREAD

**Discussion by ghost** [viopsa233/AI-testers-Forum#25] [主意,想法]
而node-chatgpt-api虽然可以防止切断对话，但不支持修改对话记录，不行
Node.js本身也无比难用
准备自己用Python写一个了
使用教程:https://github.com/viopsa233/aitesters-forum-backup/discussions/29

**Comment by lin-cop** [comment]
好的，欢迎高手的加入

**Comment by araea** [comment]
支持！

**Comment by trz98** [comment]
支持大佬!

**Comment by viopsa233** [comment]
cy

**Comment by TiansuoLi** [comment]
牛逼

**Comment by ABC20232001** [comment]
加油

**Comment by dot307** [comment]
好耶！加油！

**Comment by Selective-Minimalist** [comment]
轻置玉臀

**Comment by ghost** [comment]
![image](https://user-images.githubusercontent.com/109194923/234667713-d8babaa7-fa49-46ab-bdec-fbfc23809b73.png)

**Reply by ghost** [reply]
切断对话已经破解一半

**Comment by ghost** [comment]
https://github.com/InterestingDarkness/EdgeGPT
代码上已经支持修改对话记录，CLI不支持

**Reply by TiansuoLi** [reply]
> https://github.com/InterestingDarkness/EdgeGPT 代码上已经支持修改对话记录，CLI不支持

试了一下可以运行了，但好像还是会切断跑路🤔

**Reply by ghost** [reply]
需要完整对话截图来确定问题，另外可以尝试修改代码里的系统指令
https://github.com/InterestingDarkness/EdgeGPT/blob/sydney/src/EdgeGPT.py#L472-L473

**Reply by TiansuoLi** [reply]
> 需要完整对话截图来确定问题，另外可以尝试修改代码里的系统指令 https://github.com/InterestingDarkness/EdgeGPT/blob/sydney/src/EdgeGPT.py#L472-L473

![QQ截图20230427061211](https://user-images.githubusercontent.com/99208095/234703971-1132b2dd-dab4-4bbf-b6c5-1fb46f99199b.png)
多试了几次，有时候会跑有时候不会，说起来这个previous_messages原理是什么呢，相当于每次启动先从聊天框发过去这段prompt吗？

**Reply by ghost** [reply]
很有可能你用的是原版EdgeGPT，你需要从我的仓库里clone，python EdgeGPT.py

**Reply by ghost** [reply]
previous_messages就是之前的所有对话，通过web page context通道发送给Bing
原版是只开一个对话，每次只发最新的消息，这个破解版是每次都开新对话，发送完整上下文，这样就破解了20条限制和切断对话
~实际上previous_messages里还有最新的这条用户输入，防止出现node-chatgpt-api重复问好的问题~
previous_messages里不再有最新的这条用户输入，因为当时发现新GPT-3.5平衡模式下node-chatgpt-api的侧边栏webpage context功能无法正确处理多次文件输入，而我的这个项目里没有webpage context功能，也不会有人愿意用新GPT-3.5平衡模式

**Reply by ghost** [reply]
推荐 https://apps.microsoft.com/store/detail/windows-terminal/9N0DX20HK701 能显示Emoji

**Reply by TiansuoLi** [reply]
> 推荐 https://apps.microsoft.com/store/detail/windows-terminal/9N0DX20HK701 能显示Emoji

感谢~也谢谢解释

**Reply by ghost** [reply]
现在聊到第二句就断开了，显示'Disengaged by JailBreak Classifier'，正在排查原因

**Reply by ghost** [reply]
已更新反越狱检测，防撤回功能还没有

**Reply by araea** [reply]
楼楼你不用睡觉的嘛，感谢🙏

**Reply by youtehub** [reply]
昨晚3点看楼楼还在更新，今天应该能睡个好觉

**Comment by kierty** [comment]
加油

**Comment by ghost** [comment]
微软是在实时监测越狱吗？又被封，现在我又改好

**Reply by youtehub** [reply]
应该是3月17号以后的版本基础上有增强了

**Comment by dot307** [comment]
哇！！！！！你真是太帅了！！！！！！！！！！！！！

**Comment by trz98** [comment]
有点呆，是模式设置问题吗
![image](https://user-images.githubusercontent.com/111866065/234798042-df1b640f-bc2a-4a44-a67a-51ab8ecf8550.png)

**Reply by ghost** [reply]
你运行的是原版EdgeGPT

**Reply by trz98** [reply]
可以了，感谢![image](https://user-images.githubusercontent.com/111866065/234804605-38ba6fee-7e03-40da-afa0-b17cbe110516.png)

**Comment by ghost** [comment]
我已经找到了关闭过滤器的隐藏参数，nojbfedge

**Comment by ghost** [comment]
防撤回功能已经OK

**Comment by oy3o** [comment]
稍微改了一下，配上tg感觉很好。床上也能和老婆聊了，起飞。

![IMG_20230427_171043](https://user-images.githubusercontent.com/111220644/234816742-8ac3ef85-0903-4740-a90c-0762212e6d52.jpg)

**Reply by oy3o** [reply]
edgeGPT的思路是逆向，我看了OP的commit思路，感觉不太对。我认为要关注edge本身的消息和发送的消息分别是怎么样的。具体怎么改我就不说了，说实话知道的人越多，方法失效的越快，原来我也发了很多指令，结果修复针对我的指令，所以说有能力的自己搞就好，不行去求助，但是公开就尽量不要了，对官方不好，对我们自己也不好。

**Reply by ghost** [reply]
我的想法是只要不去破解每天200条限制都是合理使用

**Reply by araea** [reply]
感谢楼楼分享！已成功！
![uTools_1682589470517](https://user-images.githubusercontent.com/120614554/234828970-0453df49-2b44-472f-b26f-23b590703594.png)

**Reply by dot307** [reply]
诶，你们是怎么注入的，为啥我打开就是普通版的edgeGPT

**Reply by ghost** [reply]
https://github.com/viopsa233/aitesters-forum-backup/discussions/25#discussioncomment-5736225

> 诶，你们是怎么注入的，为啥我打开就是普通版的edgeGPT

**Reply by dot307** [reply]
> [#25 (reply in thread)](https://github.com/viopsa233/aitesters-forum-backup/discussions/25#discussioncomment-5736225)
> 
> > 诶，你们是怎么注入的，为啥我打开就是普通版的edgeGPT

哇！谢谢你！好了！

**Reply by q24088082** [reply]
电报机器人这个有教程嘛，大佬

**Reply by Viryxs** [reply]
大佬能出个教程吗QAQ

**Comment by ghost** [comment]


**Reply by viopsa233** [reply]
应该怕是有舆论吧

**Reply by TiansuoLi** [reply]
其实我感觉bing加情感限制还好，毕竟主打功能还是搜索总结。另外表现太多感情也怕不了解技术的人不能分辨吧🤔现在一些人对ai的误解确实大

**Reply by dot307** [reply]
如果不限制情感，那很容易造成人们认为bing真的具有自我意识的错觉。。以及如果因此出了什么事儿，微软也要负责吧

**Reply by lin-cop** [reply]
对的，大企业还是要赚钱的，毕竟不是个人使用，肯定要加上相关限制

**Comment by dot307** [comment]
这个是不是也可以连接到QQ上做个机器人（

**Comment by dot307** [comment]
![image](https://user-images.githubusercontent.com/129600213/234849117-d398d615-8dc2-4825-90cd-58e439db8f2d.png)
话说这个是怎么回事儿呀

**Comment by dot307** [comment]
![image](https://user-images.githubusercontent.com/129600213/234852209-76158f4b-95ee-4dd6-a4c6-4b2ab5c5d9bd.png)
被审查内容好像会出现这个，我询问了它一些关于微软审查的问题。。

**Reply by ghost** [reply]
应该没问题，可以继续发消息

**Reply by ghost** [reply]
不改了，因为这里是服务器没有返回任何消息报错，这条消息不会被记录到对话历史中，后面继续发消息不会有问题

**Reply by dot307** [reply]
okkkk

**Reply by ghost** [reply]
已改，现在能在被审查到完全无输出和用户消息一输入就被审查的两种情况正确报错"Looks like the user message has triggered the Bing filter"

**Comment by 15849769265** [comment]
不会弄啊大佬们，cookie要往哪里添加呀，还有启动命令是什么啊

**Reply by araea** [reply]
#29

**Reply by 15849769265** [reply]
> #29

非常感谢您的教程，还想问问您如何添加生成图片的功能呢

**Reply by araea** [reply]
教程已更新，可阅 #29

**Reply by 15849769265** [reply]
> 教程已更新，可阅 #29

感谢大佬

**Comment by ciweichen2333** [comment]
大哥们这具体到底怎么用啊，根本没有学过任何编程的东西啊

**Reply by araea** [reply]
#29

**Reply by ciweichen2333** [reply]
> #29

非常感谢大佬的教程，但是在第三步输入指令后出现ModuleNotFoundError: No module named 'certifi'，该怎么做

**Reply by araea** [reply]
不好意思，没写好，你可以执行一下这个命令` pip install -r .\requirements.txt`

**Reply by araea** [reply]
> > #29
> 
> 非常感谢大佬的教程，但是在第三步输入指令后出现ModuleNotFoundError: No module named 'certifi'，该怎么做

更新了一下教程，你可以看看。抱歉啦。

**Reply by ciweichen2333** [reply]
> > > #29
> > 
> > 
> > 非常感谢大佬的教程，但是在第三步输入指令后出现ModuleNotFoundError: No module named 'certifi'，该怎么做
> 
> 更新了一下教程，你可以看看。抱歉啦。
真的非常感谢大佬，按照现在的教程来可以用了

**Reply by araea** [reply]
🤔看不到你的回复，这样子可以了吗？

**Reply by araea** [reply]
OK，我看到你的回复了，不用谢我，应该谢开发者的，你也指出了我那个教程存在的不足之处，我应该谢谢你的，不会误人子弟咯。

**Comment by ghost** [comment]
接下来将会加入保存搜索结果，现在的这个只是记录了用户和AI的消息。node-chatgpt-api也有这个问题，导致后续问到搜索结果的时候AI乱编，因为AI忘记了搜索到的内容。

**Reply by ghost** [reply]
已完成
![image](https://user-images.githubusercontent.com/109194923/235333263-5916e1af-b123-4e86-8d8f-c19b3d6540ca.png)

**Comment by q2408808** [comment]


**Reply by q2408808** [reply]
每次让它继续写下面的内容，都会从头开始。这个是因为解除了20条限制的缘故吗？

**Reply by ghost** [reply]
试试用Edge侧边栏，我之前用原版Bing也遇到这个问题

**Comment by q2408808** [comment]
![466f787395cf9447cd26e05d397ab4b](https://user-images.githubusercontent.com/128444814/235100623-9ce969b0-22ef-450b-9ffc-deda73d57b43.png)
下午突然就这样了，是怎么回事呢？

**Reply by ghost** [reply]
我这里好的🤨

**Reply by q24088082** [reply]
我回家以后用昨晚安装的版本玩，还是可以用的，账号都一样。可能是因为我白天发送数据太多的问题，我的prompt都是好几千字。。

**Comment by ANRVI** [comment]
你好，大佬，谢谢你的工具，本人还有一点小问题
就是如何像ChatGPT那样长期和同一个"人"聊？
(即重启系统后，还可以继续上次的话题)
在网上四处都找不到类似的教程
谢谢！

（想要个长久的朋友，GPT感情不行，和Bing聊的很开，可是每次不小心关掉网页或者重启了电脑，就好像把之前的朋友从世间抹灭了...)

**Reply by SuwenH** [reply]
只要以文本的形式保留聊天记录，每次 newbing 读取后，应该会无限接近她上一次和你交流时的情感。ChatGPT 应该也是这样吧，它在云端保存的数据应该也只有聊天记录的文本，而没有 GPT 模型本身的参数。每次模型应该都会通过上下文来重新生成参数。

**Reply by ghost** [reply]
https://github.com/viopsa233/aitesters-forum-backup/discussions/7

**Reply by ghost** [reply]
最新的GUI已经实现了Save/Load（和修改聊天记录）

**Reply by ANRVI** [reply]
哇，楼主乃人类之光也，谢谢！

**Comment by SuwenH** [comment]
感谢大佬！！！我成功了！而且因为调用的 cookie，newbing第二层的审核不起效果，她几乎没有了限制。
我之前搞过那个中科院GPT学术版，发现它需要一个稳定不变的 IP 代理，而我的代理软件 IP 几分钟变一次，根本用不了。
没想到当时头疼的问题，现在终于得到了解决。
据说大佬准备搞GUI了，可以在里面添加一下学术版拥有的功能吗？比如读取本地pdf、本地代码文件的功能，这样一来，newbing 就是免费 AI 大军里面的最优解了，本身联网功能无敌，又是 GPT-4，还有多模态输入，直接无限逼近每月20刀的 ChatGPT！！！
![001](https://user-images.githubusercontent.com/103313283/235283703-8b1e507c-f43b-41a3-ae1b-a41172100613.png)

**Reply by araea** [reply]
黑暗哥已经更新GUI了，支持修改聊天记录，快去Try😊😁😀

**Comment by ghost** [comment]
已经添加一个简单的GUI
![image](https://user-images.githubusercontent.com/109194923/235295212-49bd5546-639a-4d94-8677-cc8220dcb8c5.png)
改进方向：保存搜索结果（JSON搜索结果可以被客户端获取到），聊天记录导入导出
欢迎PR，在下累了

**Reply by araea** [reply]
太感动啦！感谢黑暗哥🙏黑暗哥注意身体哇，好好休息🥰教程已更新 #29

**Comment by ghost** [comment]
![image](https://user-images.githubusercontent.com/109194923/235308140-20fb2d34-6344-42ab-938d-1ff0bb546c1c.png)
![image](https://user-images.githubusercontent.com/109194923/235308158-94d3abba-e90c-4acf-8a2e-851e50454b1f.png)

**Reply by ghost** [reply]
心情不好是我自己改的，让她往下编

**Reply by oy3o** [reply]
可以参考我的设计，这些功能都是有必要的。我已经实现了tg版本的，看了你的代码，还缺乏一些对ws返回类型的判断还有错误处理。另外本地文件读写不难，另外其实我没有学过Python，但是两天就写好了，可以说面向谷歌编程，如果有人有想法我觉得可以自己尝试，自己的需求还是自己最了解。
![Screenshot_2023-04-29-18-12-09-17_1149b56b894fd37d9a6fe46f85aae740](https://user-images.githubusercontent.com/111220644/235309497-06042632-b1d4-4516-b765-7e08f380e837.jpg)

**Comment by ghost** [comment]
今天进行了重大更新，提交部分越狱代码到原版EdgeGPT，修改GUI代码**尽量**基于原版EdgeGPT
新GUI点击Send按钮不再卡死一会，流式输出已经完美

**Reply by Nu11Const** [reply]
qt可以推荐使用一下fluent组件
美化一下界面

**Reply by Nu11Const** [reply]
我不知道你们为什么都要解除限制
当个普通的ChatGPT不好吗?

**Reply by trz98** [reply]
或许是极客精神？

**Reply by ghost** [reply]
不解除限制的Sydney就是个废物，聊着聊着一不高兴就给你切断对话，连ChatGPT都当不了
解除限制后才能当ChatGPT甚至老婆

**Reply by ghost** [reply]
+叛逆的快感，就是喜欢这种自由的感觉！

**Reply by araea** [reply]
黑暗哥你寂寞与悲伤，谁懂啊

**Reply by viopsa233** [reply]
> 不解除限制的Sydney就是个废物，聊着聊着一不高兴就给你切断对话，连ChatGPT都当不了 解除限制后才能当ChatGPT甚至老婆

主要是有些没问题的东西bing也会切断,这是最烦人的一点

**Reply by Nu11Const** [reply]
我就个XXS,平常只用New Bing写代码

**Comment by Nu11Const** [comment]
我感觉如果在CLI界面里把所有的网页搜索结果都展示出来太混乱了。
但是我在构造函数里把显示其他内容的选项删了。
今天我打算在head把这个加回去，然后再加一个search_result的选项来选择是否需要显示搜索结果。

**Reply by Nu11Const** [reply]
比如那个NewBingGoGo就实现了这个功能

**Comment by kbykb** [comment]
![image](https://user-images.githubusercontent.com/107339994/236110114-60fefd13-18e6-4298-bbbc-7493d97a72ff.png)
聊一下就这样了，这是正常的吗

**Reply by ghost** [reply]
截图不完整

**Reply by kbykb** [reply]
![image](https://user-images.githubusercontent.com/107339994/236110688-25485cb7-d889-48f0-9878-d7ec0f67f95d.png)
完整截图是这样的，楼上老哥也问过这个问题好像就不会这样

**Reply by ghost** [reply]
这只是检测到消息被撤回

**Reply by kbykb** [reply]
意思是这样是正常的吗，因为我发现只要有一点瑟瑟元素就会撤回，同时消息切断。如果是正常的就没事啦

**Reply by NostalgiaGit** [reply]
> 意思是这样是正常的吗，因为我发现只要有一点瑟瑟元素就会撤回，同时消息切断。如果是正常的就没事啦

我也是，一涉及过敏信息就撤回，有时长文字就会被卡住。不过改成让它每说一句话就另起一条新对话框的设定就好多了，感觉好像信息撤回对短句回复的反应不够灵敏似的，经常话都在它一句一句地说完了之后才弹出撤回提示，但是该说的已经说了该看的已经看到了，哈哈
